tools/backups2datalad/zarr.py

```diff
@@ -391,10 +441,28 @@ async def register_url(self, path: str, key: str, url: str) -> None:
         self.log.info("%s: Registering URL %s", path, url)
         await self.annex.register_url(key, url)
 
+    async def get_local_checksum(self) -> str:
+        if self._local_checksum is None:
+            self._local_checksum = await self.ds.compute_zarr_checksum()
+        return self._local_checksum
+
+    def get_stored_checksum(self) -> Optional[str]:
+        try:
+            return (self.repo / CHECKSUM_FILE).read_text().strip()
+        except FileNotFoundError:
+            return None
```

Comment:
we can end up in a use-case where checksum file would be annexed, content dropped, so symlink exist but not found if accessed, please add an explicit check if not a symlink, and if it is -- `raise RuntimeError(f"annexed file {self.repo / CHECKSUM_FILE} without content")` or alike to ease troubleshooting

Reply:
Are you sure that can happen?  Zarr backups are initialized with a `.dandi/.gitattributes` file to keep everything in `.dandi/` (which includes the checksum file) out of git-annex (See line 486).

Reply:
ok, good, let's hope that never happens.